ckanext/canada/plugins.py

```diff
@@ -303,6 +304,15 @@ def before_index(self, data_dict):
         return data_dict
 
 
+    # IResourceController
+    def before_update(self, context, old_dict, new_dict):
+        """
+        Set the old resource url in the context for use in the validator: canada_guess_resource_format
+        """
+        if old_dict.get('url', None) and old_dict.get('package_id', None):  # ensure that it is a resource
+            context['old_resource_url'] = old_dict.get('url', None)
```

Comment:
this doesn't work because the whole package metadata is validated together, not one resource at a time. You can grab the resource from the db to get the current value instead (we already have some validators that do this)

Reply:
Okay sounds good, just used the model orm to get the resource by id and check if it exists and has url now. All done inside of the validator method.